### DISCUSSION TITLE
Will this approach lose other features besides content filter？

### DISCUSSION THREAD

**Discussion by x9527x1024x** [eProsima/Fast-DDS#3498] [Q&A]
Hey guys,
I have a question about dynamic data structure.
Assuming we have a data type called movie, it was defined by movie.idl as below
struct movie {
    string name;
    int32 id;
    sequence<string> actor_list;
};

then create the DynamicData_ptr of movie.idl. then use the DataWriter->write(void*) to send the data.

Next，let's modify the movie.idl, put all members into an octet array.
struct movie {
    octet info[100];
};

I know, use the following structure， just have one member octet will lost the content filter features.
And, Will other features be lost?

**Comment by JLBuenoLopez** [comment]
Hi @x9527x1024x,

Dynamic Types feature allows for generating the required Type Support for serializing/deserializing the data at runtime instead of being completely defined at build time. This feature does not include the type evolution which is considered in the [XTypes specification](https://www.omg.org/spec/DDS-XTypes/1.3/About-DDS-XTypes). Currently, Fast DDS supports Dynamic Types (though with several known issues, which are [being fixed right now](https://github.com/eProsima/Fast-DDS/pull/3392)), but type evolution is not yet supported. Consequently, modifying your data type will cause to lose communication with applications that uses the first version of your data type.

Also, as you state in your comment, if you have configured a Content Filtered Topic using the TypeObject of the first data type, this filter will be invalid if the TypeObject is modified.
